### ISSUE TITLE
Include copyright notice in every source file

### ISSUE THREAD

**Issue by qingling128** [LuaJIT/LuaJIT#504]
To the repo maintainers,

Is it possible to include full copyright notice in every source file? Currently most files have something as below:

```
See Copyright Notice in lua.h
```

We have a legal requirement that all dependencies / open-source code we use need to have copyright / license. This repo seems to be the only that does not follow that practice, thus we could not use it directly.

I'm willing to send a PR if the request makes sense.

**Comment by MikuAuahDark** [comment]
Would it be something like "LuaJIT is Copyright (C) 2005-2019 Mike Pall, licensed under MIT License" is good enough to be placed into your project license or readme, additionally with MIT License text?

This idiom is actually also followed by Lua, and I think LuaJIT just use this similar pattern. So if your legal requirement requires **ALL, every single file** to have license notice, then Lua is not what you're looking for (let alone LuaJIT).

**Comment by qingling128** [comment]
Yes, that's sufficient. As long as it mentions the `MIT` license in the source file.

**Comment by qingling128** [comment]
@MikuAuahDark 
Friendly ping. Can we add `LuaJIT is Copyright (C) 2005-2019 Mike Pall, licensed under MIT License` to every file in this repo? I'm glad to contribute a PR if that approach sounds good to you.

**Comment by MikuAuahDark** [comment]
I don't know if Mike Pall himself agree. In best case, I'll wait.

Pada tanggal Kam, 12 Sep 2019 04.15, Ling Huang <notifications@github.com>
menulis:

**Comment by qingling128** [comment]
@MikePall - Hi, does https://github.com/LuaJIT/LuaJIT/issues/504#issuecomment-530547064 sound ok to you?

**Comment by qingling128** [comment]
@MikePall - Friendly ping. Should I send a PR?

**Comment by szbnwer** [comment]
hi there! :)

Mike told us that he have other duties to carry on, and slowly he just disappeared after that, and thats all we know about him currently (((maybe someone knows more, but in general this is the most that people around know)))

Siddhesh took the responsibility to fork lj and merge the pr's from here, he seems to be smart and trustworthy (if im right, then he works/worked also on gcc, so i think he have deep related knowledge), so his fork is here: https://github.com/siddhesh/LuaJIT , but i dont know how the copyright note would be appropriate in that case, so i think that u should talk with him for now, and hope that we can welcome back Mike one day, and that he is having good times meanwhile :)

good luck, and all the bests to you! :)

**Comment by qingling128** [comment]
@szbnwer - Thanks for providing the info! Sad to hear the original maintainer is no longer actively with this project. I'm a user of the `Fluent Bit` project that uses `LuaJIT` as a dependency. Let me try talk to the maintainers there first.

**Comment by szbnwer** [comment]
> Let me try talk to the maintainers there first.

right, its ur business :D (btw i just checked out fluent, its impressive :) )

**Comment by qingling128** [comment]
Yeah, it's very useful technology. Sounds like we could fix the license notice directly in Fluent Bit. :D

**Comment by szbnwer** [comment]
im not a lawyer, but i think that if the case is that finally u need to resolve this on ur own, then maybe u should fork it, rename it, and modify it somewhere before u could change the copyright stuffs, but in this case, release under ur name, otherwise it would be ambiguous that Mike hasnt done those changes that u need while u would release it under his name, and i think that wouldnt be right... and ive seen before that some licenses (like the default font of ubuntu, if im right) explicitly require to change the name, thats maybe included only because they ask it in a special way, like u r forbidden to use the original name in in the name of a fork (so like prefixing/suffixing to make difference, as thats not good for their business), but im not sure if this explicit requirement means specializations or that because this isnt a basic requirement to rename any derivated work... and now u r about to fulfill the requirements of law, so the main point is that i think these are the important bits to do that well. otherwise ive seen much forks without these done in the wild, but theres a chance that its just their lack of knowledge about the requirements of the laws... btw i think that u could search for things like "how to fork a project under X license", "how to apply X license properly" or i think that fsf or any similar organization could help u with proper info without any fees :D

bests! :)

**Comment by qingling128** [comment]
Good points. I'll consult the legal team to see what's the best way in this case.

**Comment by siddhesh** [comment]
I've been getting a bunch of notifications on this so I figured I should make my stand clear on this.  Note that I'm not a lawyer and this is merely my interpretation based on my understanding of how code has gone into luajit over the years.  Anyway, here goes:

The code copyright all over the place claims Copyright (c) Mike Pall, which is largely correct (since he has indeed written most of the code) but not entirely accurate.  There are significant bodies of code that were contributed by others (aarch64 port for example and a number of fixes) and there is no public record of a copyright assignment to Mike Pall.  The authors as a result retain copyright of that code.

That is one of the reasons why I added a CONTRIBUTORS file and added a copyright notice to indicate that the copyright owners for this source code can be seen in the CONTRIBUTORS file.  It is not complete (because I'll need to do some boring busy work to go through all of the logs and list out all of the contributors) but that IMO is the way forward for the copyright ownership mentions.

Of course, all of this changes if Mike Pall or the authors in question come forward and confirm that there was a copyright assignment in place, but there seems to be no public indication of this.
